extensions/blocks/podcast-player/edit.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-unused-vars */
+
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	TextControl,
+	Toolbar,
+	Notice,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import { edit, queueMusic } from './icons/';
+import { isAtomicSite, isSimpleSite } from '../../shared/site-type-utils';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 10;
+
+const handleSSRError = () => {
+	return <p>{ __( 'Failed to load Block', 'jetpack' ) }</p>;
+};
+
+const PodcastPlayerEdit = ( { attributes, setAttributes } ) => {
+	// Block attributes.
+	const { url, itemsToShow } = attributes;
+
+	// State.
+	const [ editedUrl, setEditedUrl ] = useState( url || '' );
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ urlError, setUrlError ] = useState( '' );
+
+	/**
+	 * Check if the current URL of the Podcast RSS feed
+	 * is valid. If so, set the block attribute and changes
+	 * the edition mode.
+	 * This function is bound to the onSubmit event for the form.
+	 *
+	 * @param {object} event Form on submit event object.
+	 */
+	const checkPodcastLink = event => {
+		event.preventDefault();
+
+		if ( ! editedUrl ) {
+			return;
+		}
+		const isValidURL = isURL( editedUrl );
```

Comment:
Maybe I'm wrong but I think here are duplicated processing in some way with how the block definition. I mean we need to define an attributes validation object, with all attributes since if there are one missing `getValidatedAttributes()` will return `undefined` for it.
In this case, we have two attributes (clearly it will increase pretty soon):

```es6
const attributesValidation = {
	itemsToShow: {
		type: 'number',
	},
	url: {
		type: 'string',
		validator: isURL,
	},
};
```

These attributes are being declared/casted here and also in the block attributes definition. 

In fact, it takes my attention that we've already been defined the validator even in the attributes.js
file.

Reply:
I've added this validation [here](https://github.com/Automattic/jetpack/pull/14948/commits/b117b518037b2a184171c7e50ae1671d4d334998) but not totally sure if it worths tbh.